src/Acuminator/Acuminator.Analyzers/StaticAnalysis/DacPropertyAttributes/MultipleAttributesOnDacPropertyFixBase.cs

```diff
@@ -0,0 +1,171 @@
+
+			if (propertyDeclaration == null)
+				return;
+
+			SemanticModel semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken)
+																.ConfigureAwait(false);
+			if (semanticModel == null)
+				return;
+
+			CodeAction codeAction =
+				CodeAction.Create(codeActionName,
+								  cToken => RemoveAllOtherAttributesFromPropertyAsync(context.Document, root, attributeNode,
+																					  propertyDeclaration, semanticModel, 
+																					  removePredicate, cToken),
+								  equivalenceKey: codeActionName);
+
+			context.RegisterCodeFix(codeAction, context.Diagnostics);
+		}
+
+		protected abstract string GetCodeActionName();
+
+		protected abstract Func<FieldTypeAttributeInfo, bool> GetRemoveAttributeByAttributeInfoPredicate();
+
+		private Task<Document> RemoveAllOtherAttributesFromPropertyAsync(Document document, SyntaxNode root, AttributeSyntax attributeNode,
+																		 PropertyDeclarationSyntax propertyDeclaration, SemanticModel semanticModel,
+																		 Func<FieldTypeAttributeInfo, bool> removePredicate, CancellationToken cancellationToken)
+		{
+			return Task.Run(() =>
+			{
+				var rewriterWalker = new MultipleAttributesRemover(document, semanticModel, attributeNode, removePredicate, cancellationToken);
+				var propertyModified = rewriterWalker.Visit(propertyDeclaration) as PropertyDeclarationSyntax;
+
+				cancellationToken.ThrowIfCancellationRequested();
+
+				var modifiedRoot = root.ReplaceNode(propertyDeclaration, propertyModified);
+				return document.WithSyntaxRoot(modifiedRoot);
+
+			}, cancellationToken);
+		}
+
+
+
+		private class MultipleAttributesRemover : CSharpSyntaxRewriter
+		{
+			private int _visitedAttributeListCounter;
+			private int _attributeListsRemovedCounter;
+			
+			private readonly Document _document;
+			private readonly SemanticModel _semanticModel;
+			private readonly FieldTypeAttributesRegister _attributesRegister;
+			private readonly AttributeSyntax _remainingAttribute;
+			private readonly Func<FieldTypeAttributeInfo, bool> _attributeToRemovePredicate;
+			private readonly CancellationToken _cancellationToken;
+
+			public MultipleAttributesRemover(Document aDocument, SemanticModel aSemanticModel, AttributeSyntax aRemainingAttribute,
+											 Func<FieldTypeAttributeInfo, bool> attributeToRemovePredicate, CancellationToken cToken)
+			{
+				_document = aDocument;
+				_semanticModel = aSemanticModel;
+				_cancellationToken = cToken;
+				_remainingAttribute = aRemainingAttribute;
+				_attributeToRemovePredicate = attributeToRemovePredicate;
+
+				PXContext pxContext = new PXContext(_semanticModel.Compilation);
+				_attributesRegister = new FieldTypeAttributesRegister(pxContext);
+			}
+
+            public override SyntaxNode VisitAttributeList(AttributeListSyntax attributeListNode)
+			{
+				_cancellationToken.ThrowIfCancellationRequested();
+
+				if (_visitedAttributeListCounter < Int32.MaxValue)
```

Comment:
Are you sure that this value should be so great?

Reply:
@dropsonic no, but it is a safety check

Reply:
@SENya1990 I mean, it will be safer to decrease it to something meaningful like 100

Reply:
I don't know - from one side it makes sense since I doubt anyone uses 100 attributes on a field including tools that autogenerate code.
 On the other side this is just a technical safety check to prevent overflow - and it will set some constraints for code which won't be ever documented.

Reply:
Technically, Int.MaxValue is a constraint which isn't documented :) I just think that `StackOverflowException` will be thrown way too earlier than this limit is achieved.

Reply:
ok, you convinced me)